openedx/core/djangoapps/waffle_utils/tests/test_views.py

```diff
@@ -33,19 +36,90 @@ def test_failure_for_non_staff(self):
         self.assertIn('waffle_flags', response.data)
         self.assertTrue(response.data['waffle_flags'])
-        # This is no longer the first flag
-        #self.assertEqual(response.data['waffle_flags'][0]['name'], 'test.flag')
+        waffle_names = [waffle["name"] for waffle in response.data['waffle_flags']]
+        self.assertIn('test.flag', waffle_names)
 
     @override_switch('test.switch', True)
     def test_response_with_waffle_switch(self):
-        response = self._get_toggle_state_response(is_staff=True)
+        response = self._get_toggle_state_response()
         self.assertIn('waffle_switches', response.data)
         self.assertTrue(response.data['waffle_switches'])
-        # This is no longer the first switch
-        #self.assertEqual(response.data['waffle_switches'][0]['name'], 'test.switch')
+        waffle_names = [waffle["name"] for waffle in response.data['waffle_switches']]
+        self.assertIn('test.switch', waffle_names)
+
+    def test_response_with_setting_toggle(self):
+        _toggle = SettingToggle("MYSETTING", default=False, module_name="module1")
+        with override_settings(MYSETTING=True):
+            response = self._get_toggle_state_response()
+
+        self.assertIn(
+            {
+                "name": "MYSETTING",
+                "is_active": True,
+                "module": "module1",
+                "class": "SettingToggle",
+            },
+            response.data["django_settings"],
+        )
+
+    def test_response_with_existing_setting_dict_toggle(self):
+        response = self._get_toggle_state_response()
+        self.assertIn(
+            {
+                "name": "FEATURES['MILESTONES_APP']",
+                "is_active": True,
+                "module": "util.milestones_helpers",
+                "class": "SettingDictToggle",
+            },
+            response.data["django_settings"],
+        )
+
+    def test_response_with_new_setting_dict_toggle(self):
+        _toggle = SettingDictToggle(
+            "CUSTOM_FEATURES", "MYSETTING", default=False, module_name="module1"
+        )
+        with override_settings(CUSTOM_FEATURES={"MYSETTING": True}):
+            response = self._get_toggle_state_response()
+
+        setting_dict = {toggle["name"]: toggle for toggle in response.data["django_settings"]}
+
+        self.assertEqual(
+            {
+                "name": "CUSTOM_FEATURES['MYSETTING']",
+                "is_active": True,
+                "module": "module1",
+                "class": "SettingDictToggle",
+            },
+            setting_dict["CUSTOM_FEATURES['MYSETTING']"],
+        )
+
+    def test_setting_overridden_by_setting_toggle(self):
+        toggle2 = SettingToggle(
+            "MYSETTING2", module_name="module1"
+        )
+        toggle3 = SettingDictToggle(
+            "MYDICT", "MYSETTING3", module_name="module1"
+        )
+        with patch.object(toggle2, "is_enabled", return_value=False) as mock_toggle2_is_enabled:
```

Comment:
I found this test confusing, because this scenario is not very likely.  Under ordinary circumstances, a SettingToggle's `is_enabled` function would return the value that is stored in the settings, which would be True in this case.

I feel like it would be a simpler and clearer test if we didn't patch the toggles, and instead if we simply had an assertion that the response list contains exactly 3 settings, the values should match the settings, and maybe test that the "class" field appears on the correct ones.

Reply:
Sure, I can simplify this test, which is admittedly convoluted.

Reply:
While rewriting this test, I discovered that setting flags added by `_add_settings` do not have a "class" entry. Is this the desired behaviour?